starfyre/__main__.py

```diff
@@ -57,26 +110,29 @@ def main(path, build):
 
     # Convert path to absolute path
     absolute_path = Path(path).resolve()
+    print(f'Absoulte path = {absolute_path}')
 
     if build:
         # Compile and build project
         init_file_path = absolute_path / "__init__.py"
         # Note: The routes specified in the pages folder will have generated code in the build directory.
         compile(init_file_path.resolve())
-     
+
         # At this point, the project has been compiled and the build directory has been created.
         # Now, initialize the Router object and use it to handle file-based routing.
-        # Basically, generate the html files for the routes
+        # Basically, get all the file names from the "pages" directory
         file_router = FileRouter(absolute_path / "pages")
-        file_router.generate_routes()
+        routes = file_router.populate_router()
+
+        # We have to create the main file.
+        # The main file will be used to generate the HTML output for all routes found by the FileRouter, index route inclusively.
+        build_routes_output(generated_routes=routes, path=str(absolute_path))
 
-        # But there is no main file in the build directory.
-        create_main_file(str(absolute_path))
         # Start/run project
         subprocess.run(
             [sys.executable, "-m", "build"],
             cwd=path,
-            stdout=subprocess.PIPE,
+            # stdout=subprocess.PIPE,
             stderr=None,
         )
 
```

Comment:
I have some changes on the main for this. Can you fetch the latest changes from the main branch here?